lib/resources/orders.js

```diff
@@ -391,6 +391,83 @@ const updateOrderItemsInContext = orderContext => {
   };
 };
 
+/**
+ * Cancel an order function.
+ *
+ * @typedef module:resources/orders~CancelFunc
+ *
+ * @property {string} orderOd order ID.
+ * @property {object} [params] [Request]{@link https://github.com/request/request} params to pass to the API call.
+ * @property {module:api~requestCallback} callback Callback that handles the response.
+ *
+ * @example
+ * // Make a post request to the API for the order with the ID of 'abc123' for program 12
+ * ro.program(12).order.cancel({
+ *  orderId: 'abc123',
+ *  refundReasonDescription: 'reason',
+ *  params: {
+ *    // params
+ *  },
+ *  callback: (error, data) => {
+ *     // ...
+ *  }
+ * });
+ */
+
+/**
+ * Higher order function for creating a `cancelOrder` function in context.
+ *
+ * @param {module:resources/orders~OrderContext} orderContext Order context object
+ *
+ * @returns {module:resources/orders~CancelFunc} `cancelOrder` function in context
+ *
+ * @see {@link module:resources/orders~CancelFunc} for examples.
+ *
+ * @protected
+ */
+const cancelOrder = orderContext => {
+  return function cancel({ orderId, refundReasonDescription, params, callback }) {
+    let error;
+    let errorMessage;
+
+    // // Argument validations.
+    if (typeof params !== 'object') {
+      // Validate that `params` exists and is an object.
+      errorMessage = 'A params object is required';
+    } else if (config.get('apiVersion') === 'v4' && (!orderId || typeof orderId !== 'string')) {
+      // Validate that `orderId` exists.
+      // NOTE: This is only required in RewardOps API v4
+      errorMessage = 'must pass an orderId to `orders.cancel()`';
+    } else if (
+      config.get('apiVersion') === 'v4' &&
+      (!refundReasonDescription || typeof refundReasonDescription !== 'string')
+    ) {
+      // Validate that `refundReasonDescription` exists.
+      // NOTE: This is only required in RewardOps API v4
+      errorMessage = 'must pass an refundReasonDescription param to `orders.cancel()`';
+    }
+
+    if (errorMessage) {
+      error = new Error();
+      error.message = errorMessage;
+      if (arguments.length === 1 && typeof params === 'function') {
+        callback = params;
+      }
+
+      callback(error);
+      return;
+    }
+
+    const options = {
+      path: `/${orderContext.contextTypeName}/${orderContext.contextId}/orders/${orderId}/refunds?refund_reason_description=${refundReasonDescription}`,
+      config: config.getAll(),
+      params,
```

Comment:
could you explain a bit better what kind of validation I should add? `refund_reason_description` is already being handle by the `refundReasonDescription` params of the call. `order_suppliers` is the body of the call and it's represented by `params` on the function. So would you mind giving me an example of the type of validation you have in mind?

Reply:
My bad, I missed `refundReasonDescription`.

My question is, will the call fail if say `supplier_id` or `fulfillment` is missing?

If so, using a schema will fail early (which is preferred IMO).

Reply:
Thanks for the help to add the schema. I update the code to use that as validation of the body, and I add everything as mandatory since it looks like the call will fail if we don't send a part of the body 
<img width="1153" alt="Screen Shot 2021-01-14 at 4 47 09 PM" src="https://user-images.githubusercontent.com/26459443/104653283-5cf47e00-5688-11eb-8f30-d0f15eb762fe.png">